### PR TITLE
feat(security): add negotiation class

### PR DESCRIPTION
**Pull request by levy5307**
1. add a configuration of `enable_auth`
2. add negotiation class
3. rename `InterceptorManger` to `InterceptorManager`

### CODE REVIEW

File: src/main/java/com/xiaomi/infra/pegasus/client/ClientOptions.java [hunk 1]
```diff
@@ -310,6 +322,17 @@ public Builder metaQueryTimeout(Duration metaQueryTimeout) {
       return this;
     }
 
+    /**
+     * Whether to enable. Defaults to {@literal false}, see {@link #DEFAULT_ENABLE_AUTH}.
```
Comment (Shuo-Jia) [inline src/main/java/com/xiaomi/infra/pegasus/client/ClientOptions.java:326]: Whether to enable authentication

File: src/main/java/com/xiaomi/infra/pegasus/rpc/async/ReplicaSession.java [hunk 2]
```diff
@@ -207,6 +209,17 @@ public void operationComplete(ChannelFuture channelFuture) throws Exception {
     }
   }
 
+  private void startNegotiation(Channel activeChannel) {
```
Comment (Shuo-Jia) [inline src/main/java/com/xiaomi/infra/pegasus/rpc/async/ReplicaSession.java:212]: StartNegotiate

Comment (levy5307) [reply src/main/java/com/xiaomi/infra/pegasus/rpc/async/ReplicaSession.java:212]: I think startNegotiation is ok. negotiate is a verb.

**Comment by neverchanje** [comment]
What does it mean "enable_auth"? Does client have the choice to skip authentication?

**Comment by levy5307** [comment]
> What does it mean "enable_auth"? Does client have the choice to skip authentication?

If server's configuration of enable_auth = true,  so client can't skip it.  Or client can skip it.

**Review by Shuo-Jia — APPROVED** [review]

**Review by acelyc111 — APPROVED** [review]

**Pull request merged**